### PR TITLE
Event/hcai/getevent blocked info

### PR DESCRIPTION
**Pull request by TCbacon**

### CODE REVIEW

File: APILambda/events/getEventById.ts [hunk 1]
```diff
@@ -42,12 +80,20 @@ export const getEventByIdRouter = (
...
     { pathParamsSchema: eventRequestSchema },
-    (req, res) => getEventById(
-      conn,
-      Number(req.params.eventId),
-      res.locals.selfId
-    )
-      .mapFailure((error) => res.status(404).json({ error }))
-      .mapSuccess((event) => res.status(200).send(event))
+    (req, res) =>
+      getEventById(conn, Number(req.params.eventId), res.locals.selfId)
```
Comment (TCbacon) [inline APILambda/events/getEventById.ts]: add transaction for making it atomic

File: APILambda/events/getEventById.ts [hunk 2]
```diff
@@ -3,13 +3,51 @@ import { z } from "zod"
...
+  name: dbUser.name,
+  title: eventTitle
+})
+
+const getUserToHostRelationWithId = (
+  conn: SQLExecutable,
+  userId: string,
+  fromUserId: string
+) => {
+  return conn.queryFirstResult<DatabaseUserToHostRelation>(
```
Comment (TCbacon) [inline APILambda/events/getEventById.ts]: use relationtouserwithid from shared file instead

File: APILambda/events/getEventById.test.ts [hunk 3]
```diff
@@ -35,3 +40,60 @@ describe("GetSingleEvent tests", () => {
...
+    const resp = await callGetEvent(attendeeToken, createEventResponse.body.id)
+
+    expect(resp).toMatchObject({
+      status: 403
+    })
+
+    expect(resp.body).toEqual({
+      name: hostName,
+      title: testEvent.title
+    })
```
Comment (TCbacon) [inline APILambda/events/getEventById.test.ts:100]: combine status and body into one expect statement

File: APILambda/events/getEventById.ts [hunk 4]
```diff
@@ -3,13 +3,51 @@ import { z } from "zod"
...
 
-export const getEventById = (conn: SQLExecutable, eventId: number, userId: string) => conn.queryFirstResult<DatabaseEvent>(
-  `
+const BlockedEventResponse = (
+  dbUser: DatabaseUserToHostRelation,
+  eventTitle: string
+) => ({
+  name: dbUser.name,
+  title: eventTitle
+})
```
Comment (TCbacon) [inline APILambda/events/getEventById.ts:24]: include handle, and profile img in blocked event response

File: APILambda/events/getEventById.test.ts [hunk 5]
```diff
@@ -35,3 +40,60 @@ describe("GetSingleEvent tests", () => {
...
+      name: hostName,
+      userId: hostUserId
+    } = await createUserFlow()
+    const { token: attendeeToken } = await createUserFlow()
+
+    await callBlockUser(attendeeToken, hostUserId)
+
+    const startTimestamp = new Date("2050-01-01")
+    const endTimestamp = new Date("2050-01-02")
+    const createEventResponse = await callCreateEvent(hostToken, {
```
Comment (TCbacon) [inline APILambda/events/getEventById.test.ts]: update tests to use createEventFlow()

**Review by seanim0920 — APPROVED** [review]

**Pull request merged**